authorization/invitation/service/invitation_service_blackbox_test.go

```diff
@@ -34,619 +39,869 @@ func (s *invitationServiceBlackBoxTest) SetupTest() {
 
-func (s *invitationServiceBlackBoxTest) TestRescindUnprivilegedInvitationFailsForInvitationID() {
-	// Create a test user - this will be the team admin
-	teamAdmin := s.Graph.CreateUser()
+		// Create a team
+		team := s.Graph.CreateTeam()
 
-	// Create a team
-	team := s.Graph.CreateTeam()
+		// Create another test user - we will invite this one to join the team
+		invitee := s.Graph.CreateUser()
 
-	// Create another test user - we will invite this one to join the team
-	invitee := s.Graph.CreateUser()
+		inv := s.Graph.CreateInvitation(team, invitee, true)
 
-	inv := s.Graph.CreateInvitation(team, invitee, true)
+		// when
+		err := s.Application.InvitationService().Rescind(s.Ctx, teamAdmin.IdentityID(), uuid.NewV4())
 
-	err := s.Application.InvitationService().Rescind(s.Ctx, teamAdmin.IdentityID(), inv.Invitation().InvitationID)
-	require.Error(s.T(), err, "Error rescinding invitation")
-	require.IsType(s.T(), errors.ForbiddenError{}, err)
+		// then
+		require.Error(t, err, "Error rescinding invitation")
+		require.IsType(t, errors.NotFoundError{}, err)
 
-	_, err = s.Application.InvitationRepository().Load(s.Ctx, inv.Invitation().InvitationID)
-	require.NoError(s.T(), err)
-}
+		// when
+		_, err = s.Application.InvitationRepository().Load(s.Ctx, inv.Invitation().InvitationID)
 
-func (s *invitationServiceBlackBoxTest) TestRescindInvitationFailsForInvalidInvitationID() {
-	// Create a test user - this will be the team admin
-	teamAdmin := s.Graph.CreateUser()
+		// then
+		require.NoError(t, err)
+	})
+	s.T().Run("should rescind invitation for resource", func(t *testing.T) {
+		// given
+		// Create a test user - this will be the team admin
+		teamAdmin := s.Graph.CreateUser()
 
-	// Create a team
-	team := s.Graph.CreateTeam()
+		// Create resource space
+		space := s.Graph.CreateSpace()
+		space.AddAdmin(teamAdmin)
 
-	// Create another test user - we will invite this one to join the team
-	invitee := s.Graph.CreateUser()
+		// Create another test user - we will invite this one to join space
+		invitee := s.Graph.CreateUser()
 
-	inv := s.Graph.CreateInvitation(team, invitee, true)
+		inv := s.Graph.CreateInvitation(space, invitee, true)
 
-	err := s.Application.InvitationService().Rescind(s.Ctx, teamAdmin.IdentityID(), uuid.NewV4())
-	require.Error(s.T(), err, "Error rescinding invitation")
-	require.IsType(s.T(), errors.NotFoundError{}, err)
+		// when
+		err := s.Application.InvitationService().Rescind(s.Ctx, teamAdmin.IdentityID(), inv.Invitation().InvitationID)
 
-	_, err = s.Application.InvitationRepository().Load(s.Ctx, inv.Invitation().InvitationID)
-	require.NoError(s.T(), err)
-}
-
-func (s *invitationServiceBlackBoxTest) TestRescindInvitationOKForResource() {
-	// Create a test user - this will be the team admin
-	teamAdmin := s.Graph.CreateUser()
+		// then
+		require.NoError(t, err, "Error rescinding invitation")
 
-	// Create resource space
-	space := s.Graph.CreateSpace()
-	space.AddAdmin(teamAdmin)
+		// when
```

Comment:
actually, the code here also belongs to the `// then` part of your test, since you're making calls to verify that some data was not inserted in the DB.